src/cpp/rtps/history/CacheChangePool.cpp

```diff
@@ -133,13 +136,40 @@ bool CacheChangePool::reserve_Cache(CacheChange_t** chan, uint32_t dataSize)
                 *chan = nullptr;
                 return false;
             }
-
             break;
 
         case DYNAMIC_RESERVE_MEMORY_MODE:
             *chan = allocateSingle(dataSize); //Allocates a single, empty CacheChange. Allocated on Copy
             if(*chan == nullptr) return false;
             break;
+
+        case DYNAMIC_REUSABLE_MEMORY_MODE:
+            if(m_freeCaches.empty())
+            {
+                *chan = allocateSingle(dataSize); //Allocates a single, empty CacheChange, only if none free
+                if(*chan == nullptr)
+                {
+                    return false;
+                }
+            }
+            else
+            {
+                *chan = m_freeCaches.back();
+                m_freeCaches.erase(m_freeCaches.end()-1);
```

Comment:
`*chan = m_freeCaches.pop_back()`

May be other parts of the code may also be modified

Reply:
done and reviewed on the rest of the document